index.html

```diff
@@ -1401,6 +1419,52 @@ <h2>Handling splits</h2>
         LoopEnd --> LoopStart
     </div>
 
+    <h2>Handling check points</h2>
+    <div class=description>
+        <p>Check point to deal with growing Parsec. Effectively creating a new section with a new Parsec but same members as before.<br />
+        </p>
+        <button class="collapsible">WAITED_VOTES</button>
+        <div class="content">
+            <p>Wait for the vote to reach consensus before reflecting the status change in our chain.<br>
+            </p>
+        </div>
+        <button class="collapsible">complete_check_point</button>
+        <div class="content">
+            <p>With the NewSectionInfo in hands, completing the check point process by joining the new section.
+            </p>
+        </div>
+    </div>
+    <div class="mermaid">
+        graph TB
+        ProcessCheckPoint["ProcessCheckPoint<br />(shared state)"]
+        style ProcessCheckPoint fill:#f9f,stroke:#333,stroke-width:4px
+
+        EndRoutine["End of ProcessCheckPoint<br />(shared state)"]
+        style EndRoutine fill:#f9f,stroke:#333,stroke-width:4px
+
+        ProcessCheckPoint --> VoteNewSections
+        VoteNewSections["vote_for(Parsec::NewSectionInfo) for the new section with a restarted Parsec instance<br /><br />WAITED_VOTES.insert(vote)"]
```

Comment:
Do we need to vote for a new section? It should be the same section? no?
Is that because the version changed?

I'm wondering whether we would really want that to affect the identify that the neighbour know us as. 
Maybe we would want a check_point version (used by routing to send message to the correct parsec in parsec_map, but not advertized to our neighbours?

Reply:
> Do we need to vote for a new section? It should be the same section? no?
Is that because the version changed?

My initial thought was yep, just bump the version

> I'm wondering whether we would really want that to affect the identify that the neighbour know us as.
> Maybe we would want a check_point version (used by routing to send message to the correct parsec in parsec_map, but not advertized to our neighbours?

Well I was operating under the assumption that splits/merges would be more common than check pointing anyways, making the overhead of altering our identity small. That said, I might be mistaken? Especially with the new thinking of using relocation rather than merges to balance the network I'm now not so sure about this anymore. Hmm, maybe I'll change to use a `check_point` version after all...

Another aspect is if the check pointing should be aborted as soon as split/merge are initiated (`split_needed` or `merge_needed` is true)?

Reply:
> Well I was operating under the assumption that splits/merges would be more common than check pointing anyways

Reply:
> Well I was operating under the assumption that splits/merges would be more common than check pointing anyways

No, Split and merges should very rarely happen (Split only every time the network double in size).
Checkpointing may be a daily, hourly occurrence (or more) since it depends on how much resource prasec takes.

> Another aspect is if the check pointing should be aborted as soon as split/merge are initiated (split_needed or merge_needed is true)?

I don't think we need to cater for that: these should rarely happen if ever, and we have time to process them because of a large buffer of adults.

Reply:
> Hmm, maybe I'll change to use a check_point version after all
BTW this is just an idea i threw in seeing your design, I did not think about it much, so it may not be the right way. But if you see it is working, it might be just fine with it.

Reply:
> No, Split and merges should very rarely happen (Split only every time the network double in size).
Checkpointing may be a daily, hourly occurrence (or more) since it depends on how much resource prasec takes.

Yes your comment made me think a little about it and I came to the same conclusion, I'll change to a check_point version scheme instead!

> I don't think we need to cater for that: these should rarely happen if ever, and we have time to process them because of a large buffer of adults.

Ok that makes it simpler :)